File: JavaScript/1915-Number of Wonderful Substrings.js
/**
 * @param {string} word
 * @return {number}
 */
var wonderfulSubstrings = function(word) {
    let dp = new Array(1025).fill(0);
    dp[0] = 1;
    let prev = 0, res = 0;
    
    for (let s of word){
        let curr = (1<<(s.charCodeAt(0) - "a".charCodeAt(0)))^prev;
        res += dp[curr];
        for (let i=0; i<10; i++){
            res += dp[curr^(1<<i)];
        }
        dp[curr] += 1;
        prev = curr;
    }
    return res;
};

File: C++/1915-Number of Wonderful Substrings.cpp
class Solution {
public:
    long long wonderfulSubstrings(string word) {
        vector<int> countState(2048, 0);
        countState[0] = 1;
        int kMod = 1e9+7;

        long long res = 0;
        int currState = 0;
        
        for (char w:word) {
            currState = currState^(1<<(w-'a'));
            res += countState[currState];
            for (int i=0; i<10; i++) {
                res += countState[currState^(1<<i)];
            }
            countState[currState]++;
        }

        return res;
    }
};